docs/httplug/index.md

```diff
@@ -0,0 +1,113 @@
+    Once it is out, we will use this PSR in the main client and deprecate the old contract.
+</p>
+
+
+See the [tutorial](tutorial.md) for a concrete example.
+
+
+## HTTPlug implementations
+
+HTTPlug implementations typically are either HTTP clients of their own, or adapters wrapping existing clients
+like Guzzle 6. In the latter case, they will depend on the required client implementation,
+so you only need to require the adapter and not the actual client.
+
+
+There are two kind of implementations:
+
+ - [php-http/client-implementation](https://packagist.org/providers/php-http/client-implementation):
+ the synchronous implementation that waits for the response / error before returning from the `sendRequest` method.
+ - [php-http/async-client-implementation](https://packagist.org/providers/php-http/async-client-implementation):
+ the asynchronous implementation that immediately returns a `Http\Client\Promise`,
+ allowing to send several requests in parallel and handling responses later.
+
+Check links above for the full list of implementations.
+
+<p class="text-warning">
+    Note: Until HTTPlug 1.0 becomes stable, we will focus on the Guzzle6 adapter.
+</p>
+
+
+## Usage in a project
+
+When writing an application, you need to require a concrete
+[implementation](https://packagist.org/providers/php-http/client-implementation).
+
+See [virtual package](virtual-package.md) for more information on the topic of working with HTTPlug implementations.
+
+
+## Installation in a reusable package
+
+In many cases, packages are designed to be reused from the very beginning.
+For example, API clients are usually used in other packages/applications, not on their own.
+
+In these cases, they should **not rely on a concrete implementation** (like Guzzle 6),
+but only require any implementation of HTTPlug. HTTPlug uses the concept of virtual packages.
+Instead of depending on only the interfaces, which would be missing an implementation,
+or depending on one concrete implementation, you should depend on the virtual package `php-http/client-implementation`
+or `php-http/async-client-implementation`. There is no package with that name,
+but all clients and adapters implementing HTTPlug declare that they provide one of this virtual package or both.
+
+You need to edit the `composer.json` of your package to add the virtual package.
+For development (installing the package standalone, running tests),
+add a concrete implementation in the `require-dev` section to make the project installable:
+
+``` json
+...
+"require": {
+    "php-http/client-implementation": "^1.0"
+},
+"require-dev": {
+    "php-http/guzzle6-adapter": "^1.0"
+},
+...
+```
+
+For extra convenience, you can use the [Discovery Component](/components/discovery) system to free the user of your
+package from having to instantiate the client.
+You should however always accept injecting the client instance to allow the user to configure the client as needed.
+
```

Comment:
then lets link to that example. maybe add to the paragraph above `See also the [example in the discovery documentation](...).`